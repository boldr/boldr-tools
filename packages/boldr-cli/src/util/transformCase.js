const chalk = require('chalk');
const changeCase = require('change-case');

const allowedNameTransforms = {
  camelCase: true,
  constantCase: true,
  headerCase: true,
  paramCase: true,
  pascalCase: true,
  snakeCase: true,
};

/**
 * @param {String} str
 * @param {String} transFn;

 */
function transformCase(str, transFn = 'pascalCase') {
  if (!allowedNameTransforms.hasOwnProperty(transFn)) {
    const message = `Invalid transform, allowed transform functions are:\n
    ${Object.keys(allowedNameTransforms).join(', ')}]`;
    throw new Error(message);
  }
  return changeCase[transFn](str);
}

export default transformCase;
