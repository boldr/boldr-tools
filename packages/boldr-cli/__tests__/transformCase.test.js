import transformCase from '../src/util/transformCase';

describe('transformCase', () => {
  it('should throw on undefined transformation function', () => {
    expect(() => transformCase('', 'foobar')).toThrowErrorMatchingSnapshot();
  });

  it('should use the correct transform function', () => {
    const transFn = [
      'camelCase',
      'constantCase',
      'headerCase',
      'paramCase',
      'pascalCase',
      'snakeCase',
    ];
    for (const transform of transFn) {
      expect(transformCase('test this', transform)).toMatchSnapshot();
    }
  });

  it('should default to pascal case', () => {
    expect(transformCase('test this')).toMatchSnapshot();
  });
});
