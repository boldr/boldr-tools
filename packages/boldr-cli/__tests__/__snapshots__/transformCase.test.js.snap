// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformCase should default to pascal case 1`] = `"TestThis"`;

exports[`transformCase should throw on undefined transformation function 1`] = `
"Invalid transform, allowed transform functions are:

    camelCase, constantCase, headerCase, paramCase, pascalCase, snakeCase]"
`;

exports[`transformCase should use the correct transform function 1`] = `"testThis"`;

exports[`transformCase should use the correct transform function 2`] = `"TEST_THIS"`;

exports[`transformCase should use the correct transform function 3`] = `"Test-This"`;

exports[`transformCase should use the correct transform function 4`] = `"test-this"`;

exports[`transformCase should use the correct transform function 5`] = `"TestThis"`;

exports[`transformCase should use the correct transform function 6`] = `"test_this"`;
